### PR TITLE
Rebuild views when `Tooltip.contents` changes

### LINKED ISSUE #13923: `Tooltip.content` doesn't rebuild views on change
Originally reported via internal Slack channel.

### PR DESCRIPTION
**Pull request by mattpap**
fixes #13923

### CODE REVIEW

File: bokehjs/test/baselines/linux/Bug__in_issue_#13923__doesn't_allow_to_rebuild_views_when_Tooltip.contents_changes.blf [hunk 1]
```diff
@@ -0,0 +1 @@
+Tooltip bbox=[0, 0, 114, 29]
```
Comment (mattpap) [inline bokehjs/test/baselines/linux/Bug__in_issue_#13923__doesn't_allow_to_rebuild_views_when_Tooltip.contents_changes.blf:1]: Those `Tooltip` bboxes actually don't make too much sense, i.e. `y` should be non-zero in this context.

File: bokehjs/src/lib/models/ui/tooltip.ts [hunk 2]
```diff
@@ -113,7 +121,12 @@ export class TooltipView extends UIElementView {
       this.render()
       this.after_render()
     })
-    this.on_change([content, closable, interactive], () => {
+    this.on_change(content, async () => {
+      await this._build_content()
+      this.render()
+      this.after_render()
+    })
+    this.on_change([closable, interactive], () => {
```
Comment (mattpap) [inline bokehjs/src/lib/models/ui/tooltip.ts:129]: Ideally this `on_change` call would need to change and thus `render/after_render` would need duplication, but there's race condition related to async and non-async handlers preventing this.

Comment (hoxbro) [reply bokehjs/src/lib/models/ui/tooltip.ts:129]: I'm not following you here. Are you saying that if there weren't a race condition you would have to call both of these methods twice?

**Review by hoxbro — APPROVED** [review]

File: bokehjs/src/lib/models/ui/tooltip.ts (hunk 2 above)
Comment (mattpap) [reply bokehjs/src/lib/models/ui/tooltip.ts:129]: I would like to be able to write this:
```ts
    this.on_change(content, async () => {
      await this._build_content()
    })
    this.on_change([content, closable, interactive], () => {
      this.render()
      this.after_render()
    })
```
Handlers are run in registration order.

**Pull request merged**